### PR TITLE
[CDAP-20632] Add new aggregation for app entities

### PR DESCRIPTION
**Pull request by vanathi-g**
- Reverts changes in PR https://github.com/cdapio/cdap/pull/15053.
- Add a new "BY_ENTITY" aggregation with necessary tags

### CODE REVIEW

File: cdap-watchdog/src/main/java/io/cdap/cdap/metrics/store/DefaultMetricStore.java [hunk 1]
```diff
@@ -161,8 +162,7 @@ public class DefaultMetricStore implements MetricStore {
     aggs.put(BY_WORKFLOW, new DefaultAggregation(
         ImmutableList.of(Constants.Metrics.Tag.NAMESPACE, Constants.Metrics.Tag.APP,
             Constants.Metrics.Tag.WORKFLOW, Constants.Metrics.Tag.DATASET,
-            Constants.Metrics.Tag.RUN_ID, Constants.Metrics.Tag.NODE,
-            Constants.Metrics.Tag.APP_ENTITY_TYPE, Constants.Metrics.Tag.APP_ENTITY_TYPE_NAME),
+            Constants.Metrics.Tag.RUN_ID, Constants.Metrics.Tag.NODE),
```
Comment (de-lan) [inline cdap-watchdog/src/main/java/io/cdap/cdap/metrics/store/DefaultMetricStore.java]: I'd suggest to split this PR into two: one as the revert commit in https://github.com/cdapio/cdap/pull/15170, this one focuses on adding the metric aggregation.

File: cdap-watchdog/src/main/java/io/cdap/cdap/metrics/store/DefaultMetricStore.java [hunk 2]
```diff
@@ -87,6 +87,7 @@ public class DefaultMetricStore implements MetricStore {
   private static final String BY_COMPONENT = "component";
   private static final String BY_SCHEDULE = "schedule";
   private static final String BY_ONLY_COMPONENT = "only_component";
+  private static final String BY_ENTITY = "entity";
```
Comment (de-lan) [inline cdap-watchdog/src/main/java/io/cdap/cdap/metrics/store/DefaultMetricStore.java]: Just wondering does it make more sense to use `BY_APP_ENTITY = "app_entity";`?

**Review by sumengwang — APPROVED** [review]

File: cdap-watchdog/src/main/java/io/cdap/cdap/metrics/store/DefaultMetricStore.java (hunk 1 above)
Comment (vanathi-g) [reply cdap-watchdog/src/main/java/io/cdap/cdap/metrics/store/DefaultMetricStore.java]: Sure, once the other PR is merged, I'll rebase this branch and force-push

File: cdap-watchdog/src/main/java/io/cdap/cdap/metrics/store/DefaultMetricStore.java (hunk 2 above)
Comment (vanathi-g) [reply cdap-watchdog/src/main/java/io/cdap/cdap/metrics/store/DefaultMetricStore.java]: Makes sense, done

**Review by de-lan — APPROVED** [review]

**Pull request merged**